### PR TITLE
Rename num/den to numerator/denominator

### LINKED ISSUE #19233: Possible remaming of `num` and `den`?
Could `num` and `den` be renamed to `numerator` and `denominator`? These function names are very unintuitive, and general policy is to avoid abbreviations unless they're very standard.

### PR DESCRIPTION
**Pull request by fredrikekre**
Fix #19233 

Think I found all the places. Also moved docstrings inline.

### CODE REVIEW

**Review by StefanKarpinski — CHANGES_REQUESTED** [review]
Needs an entry in NEWS.md. Looks good otherwise!

**Comment by fredrikekre** [comment]
@StefanKarpinski  Link to issue or PR? Seems like all entries in `NEWS.md` are linked to the issue, but in `deprecated.jl` the references are for the PR.

**Comment by StefanKarpinski** [comment]
I don't think it really matters which as long as there's an entry and a trail of breadcrumbs.

**Review by StefanKarpinski — APPROVED** [review]

**Comment by fredrikekre** [comment]
I don't have the power of the button, but I think this is all done?

**Comment by KristofferC** [comment]
LGTM

File: NEWS.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ Deprecated or removed
 
   * `is` has been deprecated in favor of `===` (which used to be an alias for `is`) ([#17758]).
 
+  * `num` and `den` has been deprecated in favor of `numerator` and `denominator` respectively ([#19233]).
```
Comment (nalimilan) [inline NEWS.md]: has -> have

Comment (fredrikekre) [inline NEWS.md]: fixed

**Comment by StefanKarpinski** [comment]
There seems to be a modicum of disagreement on the original issue, so let's that discussion play out before merging this.

**Comment by martinholters** [comment]
I think that discussion is resolved in favor of the change. Merge?

**Comment by KristofferC** [comment]
Argh, I think I left the CI skip message in the squash merge. Sorry :/

**Comment by StefanKarpinski** [comment]
Not your fault: controlling the CI tools via commit messages is an incredibly dump, bad UX design.

**Pull request merged**